Automattic/wp-calypso
# Plans: Remove `siteSpecificPlansDetailsList` and `siteSpecificPlansDetailsMixin` in favor of new actions/reducers

This PR updates the components in `client/my-sites/plans/` to use site plans data in the global state.

Previously, the `/plans` components used `siteSpecificPlansDetailsList`, which:

- Was a module containing a singleton class that fetched/stored data related to site plans.
- Used `site.domain` instead of `site.slug` to namespace plans, which causes namespace conflicts for sites that share a `domain` property.
- Was used with a mixin, `siteSpecificPlansDetailsMixin`.

This PR:

- Removes `siteSpecificPlansDetailsList` and `siteSpecificPlansDetailsMixin`.
- Updates all instances of use of the former to use the new global store.

**Testing**
The plans components are used in both `/plans` and on the plans step in signup, so it is important that we assert that the behavior of both places remains unchanged.

*Signup*
- Visit http://calypso.localhost:3000/start
  - Assert that you can complete signup with the free plan, and are not redirected to checkout afterward.
  - Assert that you can complete signup with a paid plan, and are redirected to checkout afterward.
  - Assert that you can view the 'Compare Plans' page on the plans step by clicking one of the 'Learn more' links.

*Plans*
- Visit http://calypso.localhost:3000/plans/:site where `:site` is the slug of a site with the free plan.
  - Assert that you are taken to checkout with a free trial by clicking 'Start Free Trial'.
  - Assert that you are taken to checkout with a full plan by clicking 'Upgrade Now'.
- Visit http://calypso.localhost:3000/plans/:site where `:site` is the slug of a site with a free trial.
  - Assert that you see the `PlanOverview` component indicating the amount of time remaining for the trial.
  - Assert that you are taken to checkout with the full plan by clicking 'Purchase Now'.
- Visit http://calypso.localhost:3000/plans/:site where `:site` is the slug of a site with a premium/business plan.
  - Assert that the behavior of the buttons below each plan (to manage the plan or upgrade to business if the site has premium) is unchanged.

**Reviews**
